iOS/Controllers/BottomSheetViewController.cs

```diff
@@ -34,184 +33,200 @@ public sealed class BottomSheetViewController : UIViewController
 
-            var blurView = new UIVisualEffectView(ApplicationTheme.PanelBackgroundMaterial)
-            {
-                TranslatesAutoresizingMaskIntoConstraints = false,
-                ClipsToBounds = true
-            };
+        /// <summary>
+        /// Anchor to use for constraining views (e.g. attribution) to the top of this panel when in compact width (bottom sheet) mode.
+        /// </summary>
+        public NSLayoutYAxisAnchor PanelTopAnchor { get; private set; }
 
-            // Defined in Helpers/ViewExtensions
-            var blurShadowContainerView = blurView.EncapsulateInShadowView();
-            View.AddSubview(blurShadowContainerView);
+        /// <summary>
+        /// Defines the height to use when in the partial state and the height of the content can't be determined.
+        /// </summary>
+        public nfloat DefaultPartialHeight { get; set; } = 160;
 
-            DisplayedContentView.BackgroundColor = UIColor.Clear;
-            DisplayedContentView.ClipsToBounds = true;
+        /// <summary>
+        /// Defines the size of the content view when the view state is minimized.
+        /// Generally only used if <see cref="AllowsMinimizedState"/> or <see cref="AllowsManualResize"/> is <value>true</value>.
+        /// </summary>
+        public nfloat MinimumHeight { get; set; } = 80;
+
+        /// <summary>
+        /// Determines if the view can be set to the Minimized state. If <value>false</value>, the partial state is used in place of minimized.
+        /// </summary>
+        public bool AllowsMinimizedState { get; set; } = false;
+
+        /// <summary>
+        /// If <value>true</value>, the user can pan to adjust the size of the view. When <value>true</value>,
+        /// a handlebar is shown to indicate that the view is adjustable.
+        /// </summary>
+        public bool AllowsManualResize { get; set; } = false;
+
+        /// <summary>
+        /// Gets the maximum height of the sheet based on the current UI trait collection
+        /// </summary>
+        private nfloat MaxHeightConstraint
+        {
+            get
+            {
+                switch (TraitCollection.HorizontalSizeClass)
+                {
+                    case UIUserInterfaceSizeClass.Compact:
+                        return View.Superview.Frame.Height + ApplicationTheme.Margin - View.Superview.SafeAreaInsets.Top - (2 * ApplicationTheme.Margin);
+                    case UIUserInterfaceSizeClass.Regular:
+                    default:
+                        return View.Superview.Frame.Height - View.Superview.SafeAreaInsets.Top - (2 * ApplicationTheme.Margin) - View.Superview.SafeAreaInsets.Bottom - (2 * ApplicationTheme.Margin);
+                }
+            }
+        }
 
-            blurView.ContentView.AddSubview(DisplayedContentView);
+        public override void ViewWillAppear(bool animated)
```

Comment:
You have call `super.viewWillAppear(animated)` at some point.  I'd do it right at the top.  Same goes for all the `view.` methods you override.

https://developer.apple.com/documentation/uikit/uiviewcontroller/1621510-viewwillappear?language=swift

"If you override this method, you must call super at some point in your implementation."